### PR TITLE
Draft: QNN refactoring

### PR DESCRIPTION
**Pull request by David-Kreplin**
Branch for refactoring the low-level QNN implementation to make it more modular:

- [ ] Create QNN Base Class derived from the current low-level QNN implementation (@David-Kreplin)
- [ ] Adjust existing low-level QNN program to the Base Class (@David-Kreplin )
- [ ] Modifie Loss functions to return Loss and Gradient computed from the QNN(@MoritzWillmann)
- [ ] Adapt training.py to the changes in the Loss function (@MoritzWillmann)
- [ ] Add x=QuantumCircuit support to the existing low-level QNN implementation (@MHagelueken and @David-Kreplin)

### CODE REVIEW

**Comment by David-Kreplin** [comment]
Not needed anymore
